x/configurl/split.go

```diff
@@ -29,11 +30,48 @@ func registerSplitStreamDialer(r TypeRegistry[transport.StreamDialer], typeID st
 		if err != nil {
 			return nil, err
 		}
-		prefixBytesStr := config.URL.Opaque
-		prefixBytes, err := strconv.Atoi(prefixBytesStr)
+		prefixBytes, repeatsNumber, skipBytes, err := parseURL(config.URL.Opaque)
 		if err != nil {
-			return nil, fmt.Errorf("prefixBytes is not a number: %v. Split config should be in split:<number> format", prefixBytesStr)
+			return nil, err
 		}
-		return split.NewStreamDialer(sd, int64(prefixBytes))
+		return split.NewStreamDialer(sd, prefixBytes, repeatsNumber, skipBytes)
 	})
 }
+
+func parseURL(splitUrl string) (prefixBytes int64, repeatsNumber int64, skipBytes int64, err error) {
+	// Split the input string by commas
+	parts := strings.Split(splitUrl, ",")
```

Comment:
Instead of a comma-separated value, let's use proper URL variables.
It would read something like `prefix=1&count=2&length=10`. Need to figure out proper names.

You can fallback to variables if it fails to parse as a number.